From d8518ea605f5f4bb361e657314ee8013ca52b295 Mon Sep 17 00:00:00 2001
From: Aaron <aaronmedeiros@gmail.com>
Date: Wed, 12 Aug 2026 14:02:54 -0400
Subject: [PATCH] windows: brand application registration as mpv-sk

---
 osdep/mpv.rc         |  8 +++---
 osdep/w32_register.c | 63 ++++++++++++++++++++++----------------------
 2 files changed, 35 insertions(+), 36 deletions(-)

diff --git a/osdep/mpv.rc b/osdep/mpv.rc
index e0a20a2a2..6e322c9d1 100644
--- a/osdep/mpv.rc
+++ b/osdep/mpv.rc
@@ -30,12 +30,12 @@ VS_VERSION_INFO VERSIONINFO
         BLOCK "StringFileInfo" {
             BLOCK "000004b0" {
                 VALUE "Comments", "mpv is distributed under the terms of the GNU General Public License Version 2 or later."
-                VALUE "CompanyName", "mpv"
-                VALUE "FileDescription", "mpv"
+                VALUE "CompanyName", "mpv-sk"
+                VALUE "FileDescription", "mpv-sk media player"
                 VALUE "FileVersion", VERSION
                 VALUE "LegalCopyright", MPVCOPYRIGHT_W
-                VALUE "OriginalFilename", "mpv.exe"
-                VALUE "ProductName", "mpv"
+                VALUE "OriginalFilename", "mpv-sk.exe"
+                VALUE "ProductName", "mpv-sk"
                 VALUE "ProductVersion", VERSION
             }
         }
diff --git a/osdep/w32_register.c b/osdep/w32_register.c
index 214ce7210..117097157 100644
--- a/osdep/w32_register.c
+++ b/osdep/w32_register.c
@@ -32,8 +32,8 @@
 
 #include "windows_utils.h"
 
-#define MPV_NAME L"mpv"
-#define MPV_FRIENDLY_NAME L"mpv media player"
+#define MPV_NAME L"mpv-sk"
+#define MPV_FRIENDLY_NAME L"mpv-sk media player"
 
 #define KEY_AUTOPLAY L"Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\AutoplayHandlers"
 #define KEY_MPV_APP L"Software\\Classes\\Applications\\" MPV_NAME L".exe"
@@ -43,7 +43,7 @@
 #define KEY_MPV_CAPABILITIES KEY_MPV_CAPABILITIES_APP L"\\Capabilities"
 #define KEY_MPV_UNINSTALL L"Software\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\" MPV_NAME
 
-#define MPV_PROG_ID_PREFIX L"io.mpv."
+#define MPV_PROG_ID_PREFIX L"io.mpv-sk."
 #define MPV_PROG_ID(h) MPV_PROG_ID_PREFIX h
 #define KEY_MPV_PROG_ID(h) L"Software\\Classes\\" MPV_PROG_ID(h)
 
@@ -217,25 +217,25 @@ static wchar_t *w32_get_shortcut_path(struct mp_log *log)
 #define REGISTER_AUTOPLAY_HANDLER(media, event, action)                        \
     reg_add_str(log, root, KEY_MPV_PROG_ID(media) L"\\shell", NULL, L"open");  \
     reg_add_str(log, root, KEY_MPV_PROG_ID(media) L"\\shell\\open\\command",   \
-                NULL, L"mpv.exe " media L":// --" media L"-device=%L");        \
+                NULL, L"mpv-sk.exe " media L":// --" media L"-device=%L");     \
     reg_add_str(log, root,                                                     \
-                KEY_AUTOPLAY L"\\Handlers\\MpvPlay" event "OnArrival",         \
+                KEY_AUTOPLAY L"\\Handlers\\MpvSkPlay" event "OnArrival",       \
                 L"Action", L"" action);                                        \
     reg_add_str(log, root,                                                     \
-                KEY_AUTOPLAY L"\\Handlers\\MpvPlay" event "OnArrival",         \
+                KEY_AUTOPLAY L"\\Handlers\\MpvSkPlay" event "OnArrival",       \
                 L"DefaultIcon", icon_path);                                    \
     reg_add_str(log, root,                                                     \
-                KEY_AUTOPLAY L"\\Handlers\\MpvPlay" event "OnArrival",         \
+                KEY_AUTOPLAY L"\\Handlers\\MpvSkPlay" event "OnArrival",       \
                 L"InvokeProgID", MPV_PROG_ID(media));                          \
     reg_add_str(log, root,                                                     \
-                KEY_AUTOPLAY L"\\Handlers\\MpvPlay" event "OnArrival",         \
+                KEY_AUTOPLAY L"\\Handlers\\MpvSkPlay" event "OnArrival",       \
                 L"InvokeVerb", L"open");                                       \
     reg_add_str(log, root,                                                     \
-                KEY_AUTOPLAY L"\\Handlers\\MpvPlay" event "OnArrival",         \
+                KEY_AUTOPLAY L"\\Handlers\\MpvSkPlay" event "OnArrival",       \
                 L"Provider", MPV_FRIENDLY_NAME);                               \
     reg_add_str(log, root,                                                     \
                 KEY_AUTOPLAY L"\\EventHandlers\\Play" event "OnArrival",       \
-                L"MpvPlay" event "OnArrival", L"");
+                L"MpvSkPlay" event "OnArrival", L"");
 
 static void w32_register(struct MPContext *mpctx)
 {
@@ -255,7 +255,7 @@ static void w32_register(struct MPContext *mpctx)
     icon_path[mpv_path_len + 2] = L'\0';
 
     HKEY root = is_admin() ? HKEY_LOCAL_MACHINE : HKEY_CURRENT_USER;
-    mp_msg(log, MSGL_INFO, "Registering mpv for %s...\n",
+    mp_msg(log, MSGL_INFO, "Registering mpv-sk for %s...\n",
            root == HKEY_LOCAL_MACHINE ? "all users" : "the current user");
 
     // Register mpv as an application
@@ -271,13 +271,12 @@ static void w32_register(struct MPContext *mpctx)
     }
 
     // Name of the "Open With" entry in the context menu
-    // I prefer a simple "mpv" here, so let's skip this.
-    // reg_add_str(log, root, KEY_MPV_APP, L"FriendlyAppName", MPV_FRIENDLY_NAME);
+    reg_add_str(log, root, KEY_MPV_APP, L"FriendlyAppName", MPV_NAME);
 
     // Add shell open command
     reg_add_str(log, root, KEY_MPV_APP L"\\shell", NULL, L"open");
     reg_add_str(log, root, KEY_MPV_APP L"\\shell\\open\\command", NULL,
-                L"mpv.exe -- \"%L\"");
+                L"mpv-sk.exe -- \"%L\"");
 
     // Register mpv capabilities and handlers
 
@@ -290,13 +289,13 @@ static void w32_register(struct MPContext *mpctx)
     // duplicate this for each supported protocol or file type.
 
     // Register URL handler
-    reg_add_str(log, root, KEY_MPV_PROG_ID("url"), NULL, L"URL:mpv");
+    reg_add_str(log, root, KEY_MPV_PROG_ID("url"), NULL, L"URL:mpv-sk");
     reg_add_dwr(log, root, KEY_MPV_PROG_ID("url"), L"EditFlags", FTA_Show);
     reg_add_str(log, root, KEY_MPV_PROG_ID("url"), L"FriendlyTypeName",
-                L"mpv URL handler");
+                L"mpv-sk URL handler");
     reg_add_str(log, root, KEY_MPV_PROG_ID("url") L"\\shell", NULL, L"open");
     reg_add_str(log, root, KEY_MPV_PROG_ID("url") L"\\shell\\open\\command",
-                NULL, L"mpv.exe -- \"%L\"");
+                NULL, L"mpv-sk.exe -- \"%L\"");
     reg_add_str(log, root, KEY_MPV_PROG_ID("url") L"\\shell\\open\\ddeexec",
                 NULL, L"");
 
@@ -337,12 +336,12 @@ static void w32_register(struct MPContext *mpctx)
     }
 
     // Register file handler
-    reg_add_str(log, root, KEY_MPV_PROG_ID("file"), NULL, L"mpv");
-    reg_add_str(log, root, KEY_MPV_PROG_ID("file"), L"FriendlyTypeName", L"mpv Media File");
+    reg_add_str(log, root, KEY_MPV_PROG_ID("file"), NULL, L"mpv-sk");
+    reg_add_str(log, root, KEY_MPV_PROG_ID("file"), L"FriendlyTypeName", L"mpv-sk media file");
     reg_add_dwr(log, root, KEY_MPV_PROG_ID("file"), L"EditFlags", FTA_OpenIsSafe);
     reg_add_str(log, root, KEY_MPV_PROG_ID("file") L"\\shell", NULL, L"open");
     reg_add_str(log, root, KEY_MPV_PROG_ID("file") L"\\shell\\open\\command",
-                NULL, L"mpv.exe -- \"%L\"");
+                NULL, L"mpv-sk.exe -- \"%L\"");
 
     // Register file associations
     char **exts_groups[] = {
@@ -399,15 +398,15 @@ static void w32_register(struct MPContext *mpctx)
 
     // Register uninstaller
     // <https://learn.microsoft.com/windows/win32/msi/uninstall-registry-key>
-    reg_add_str(log, root, KEY_MPV_UNINSTALL, L"DisplayName", L"mpv media player");
+    reg_add_str(log, root, KEY_MPV_UNINSTALL, L"DisplayName", MPV_FRIENDLY_NAME);
     reg_add_str(log, root, KEY_MPV_UNINSTALL, L"DisplayIcon", icon_path);
     reg_add_str(log, root, KEY_MPV_UNINSTALL, L"DisplayVersion", L"" VERSION);
     reg_add_str(log, root, KEY_MPV_UNINSTALL, L"Version", L"" VERSION);
     reg_add_dwr(log, root, KEY_MPV_UNINSTALL, L"Language", 1033);
-    reg_add_str(log, root, KEY_MPV_UNINSTALL, L"UninstallString", L"mpv.exe --no-config --unregister");
+    reg_add_str(log, root, KEY_MPV_UNINSTALL, L"UninstallString", L"mpv-sk.exe --no-config --unregister");
     reg_add_dwr(log, root, KEY_MPV_UNINSTALL, L"NoModify", 1);
     reg_add_dwr(log, root, KEY_MPV_UNINSTALL, L"NoRepair", 1);
-    reg_add_str(log, root, KEY_MPV_UNINSTALL, L"Publisher", L"mpv");
+    reg_add_str(log, root, KEY_MPV_UNINSTALL, L"Publisher", L"mpv-sk");
     reg_add_str(log, root, KEY_MPV_UNINSTALL, L"HelpLink", L"https://mpv.io/manual");
     reg_add_str(log, root, KEY_MPV_UNINSTALL, L"Readme", L"https://mpv.io/community");
     reg_add_str(log, root, KEY_MPV_UNINSTALL, L"URLInfoAbout", L"https://mpv.io");
@@ -420,8 +419,8 @@ static void w32_register(struct MPContext *mpctx)
     if (SUCCEEDED(PathCchRemoveFileSpec(mpv_path, MP_PATH_MAX)))
         reg_add_str(log, root, KEY_MPV_UNINSTALL, L"InstallLocation", mpv_path);
 
-    mp_msg(log, MSGL_INFO, "mpv has been successfully registered.\n");
-    mp_msg(log, MSGL_INFO, "Note: The mpv binary has not been copied or moved. Please keep it in the current directory.\n");
+    mp_msg(log, MSGL_INFO, "mpv-sk has been successfully registered.\n");
+    mp_msg(log, MSGL_INFO, "Note: The mpv-sk binary has not been copied or moved. Please keep it in the current directory.\n");
     mp_msg(log, MSGL_INFO, "If you move it, simply run this process again to re-register.\n");
 
     talloc_free(tmp);
@@ -432,7 +431,7 @@ static void w32_unregister(struct MPContext *mpctx)
     struct mp_log *log = mp_log_new(NULL, mpctx->log, "win32");
 
     HKEY root = is_admin() ? HKEY_LOCAL_MACHINE : HKEY_CURRENT_USER;
-    mp_msg(log, MSGL_INFO, "Unregistering mpv for %s...\n",
+    mp_msg(log, MSGL_INFO, "Unregistering mpv-sk for %s...\n",
            root == HKEY_LOCAL_MACHINE ? "all users" : "the current user");
 
     reg_del(log, root, KEY_MPV_APP_PATH(".exe"), NULL);
@@ -447,15 +446,15 @@ static void w32_unregister(struct MPContext *mpctx)
     reg_del(log, root, KEY_MPV_PROG_ID("file"), NULL);
     reg_del(log, root, KEY_MPV_PROG_ID("url"), NULL);
 
-    reg_del(log, root, KEY_AUTOPLAY L"\\Handlers\\MpvPlayCDAudioOnArrival", NULL);
+    reg_del(log, root, KEY_AUTOPLAY L"\\Handlers\\MpvSkPlayCDAudioOnArrival", NULL);
     reg_del(log, root, KEY_AUTOPLAY L"\\EventHandlers\\PlayCDAudioOnArrival",
-            L"MpvPlayCDAudioOnArrival");
-    reg_del(log, root, KEY_AUTOPLAY L"\\Handlers\\MpvPlayDVDMovieOnArrival", NULL);
+            L"MpvSkPlayCDAudioOnArrival");
+    reg_del(log, root, KEY_AUTOPLAY L"\\Handlers\\MpvSkPlayDVDMovieOnArrival", NULL);
     reg_del(log, root, KEY_AUTOPLAY L"\\EventHandlers\\PlayDVDMovieOnArrival",
-            L"MpvPlayDVDMovieOnArrival");
-    reg_del(log, root, KEY_AUTOPLAY L"\\Handlers\\MpvPlayBluRayOnArrival", NULL);
+            L"MpvSkPlayDVDMovieOnArrival");
+    reg_del(log, root, KEY_AUTOPLAY L"\\Handlers\\MpvSkPlayBluRayOnArrival", NULL);
     reg_del(log, root, KEY_AUTOPLAY L"\\EventHandlers\\PlayBluRayOnArrival",
-            L"MpvPlayBluRayOnArrival");
+            L"MpvSkPlayBluRayOnArrival");
 
     wchar_t *shortcut_path = w32_get_shortcut_path(log);
     if (shortcut_path) {
-- 
2.54.0.windows.1

